src/main/resources/META-INF/plugin.xml

```diff
@@ -21,7 +21,7 @@
     <depends>org.intellij.scala</depends>
 
     <extensions defaultExtensionNs="com.intellij">
-        <backgroundPostStartupActivity implementation="fi.aalto.cs.apluscourses.intellij.activities.InitializationActivity"/>
+        <postStartupActivity implementation="fi.aalto.cs.apluscourses.intellij.activities.InitializationActivity"/>
```

Comment:
Do you have a link or something where I could read about the differences of these two? My googling isn't bringing results :smile:

Reply:
https://plugins.jetbrains.com/docs/intellij/plugin-components.html#project-open

Reply:
Thanks!

Running that initialization activity on the EDT (more familiarly the "UI thread") should be avoided, because the UI freezes until the activity is done. This is especially important in this case, because the initialization activity fetches network resources, which can have long and unpredictable delays. While running the code I occasionally got log entries like this:
```
2021-03-14 23:04:32,865 [  20349]   WARN - .diagnostic.PerformanceWatcher - UI was frozen for 12581ms, details saved to /l/drosden1/intellij-plugin/build/idea-sandbox/system/log/threadDumps-freeze-20210314-230425-IC-203.7148.57-ObservableReadWriteProperty.get-12sec
```
Then again running it in the background makes the rest more difficult. The more I work on this issue, the more I realize how complicated it is to solve it :smile: I'm personally at least tempted to reconsider whether this should be "release-blocking", let's discuss in tomorrow's daily.